bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastHandler.java

```diff
@@ -0,0 +1,1331 @@
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.StateOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link YamahaMusiccastHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Lennert Coopman - Initial contribution
+ */
+@NonNullByDefault
+public class YamahaMusiccastHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(YamahaMusiccastHandler.class);
+    private @Nullable ScheduledFuture<?> keepUdpEventsAliveTask;
+
+    private @NonNullByDefault({}) YamahaMusiccastConfiguration config;
+    private @NonNullByDefault({}) String httpResponse;
+
+    JsonParser parser = new JsonParser();
+    String tmpString = "";
+    int tmpInteger = 0;
+    int volumePercent = 0;
+    int volumeAbsValue = 0;
+    int connectionTimeout = 5000;
+    int longConnectionTimeout = 60000;
+    @Nullable
+    String responseCode = "";
+    @Nullable
+    String powerState = "";
+    @Nullable
+    String muteState = "";
+    int volumeState = 0;
+    int maxVolumeState = 0;
+    @NonNullByDefault({})
+    String inputState = "";
+    int presetNumber = 0;
+    @Nullable
+    String soundProgramState = "";
+    int sleepState = 0;
+    @NonNullByDefault({})
+    String playbackState = "";
+    @NonNullByDefault({})
+    String artistState = "";
+    @NonNullByDefault({})
+    String trackState = "";
+    @NonNullByDefault({})
+    String albumState = "";
+    @NonNullByDefault({})
+    String albumArtUrlState = "";
+    @NonNullByDefault({})
+    String repeatState = "";
+    @NonNullByDefault({})
+    String shuffleState = "";
+    @NonNullByDefault({})
+    String topicAVR = "";
+    @NonNullByDefault({})
+    String zone = "main";
+    String channelWithoutGroup = "";
+    @NonNullByDefault({})
+    String thingLabel = "";
+    @NonNullByDefault({})
+    String mclinkSetupServer = "";
+    @NonNullByDefault({})
+    String mclinkSetupZone = "";
+    String url = "";
+    String json = "";
+    String action = "";
+
+    int zoneNum = 0;
+    @NonNullByDefault({})
+    String groupId = "";
+    @NonNullByDefault({})
+    String role = "";
+    @NonNullByDefault({})
+    String host;
+    @NonNullByDefault({})
+    public String deviceId = "";
+
+    private YamahaMusiccastStateDescriptionProvider stateDescriptionProvider;
+
+    public YamahaMusiccastHandler(Thing thing, YamahaMusiccastStateDescriptionProvider stateDescriptionProvider) {
+        super(thing);
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!command.equals(RefreshType.REFRESH)) {
```

Comment:
Same for the ON/OFF commands.
```suggestion
        if (command != RefreshType.REFRESH) {
```

Reply:
This gives me a Type mismatch: cannot convert from Command to boolean. Any suggestion?

Reply:
I don't get any error...?

Reply:
My mistake, typo. It works now.